controllers/storagecluster/external_resources.go

```diff
@@ -390,22 +391,17 @@ func (r *StorageClusterReconciler) createExternalStorageClusterResources(instanc
 					return err
 				}
 			} else if d.Name == cephRgwStorageClassName {
-				rgwEndpoint := d.Data[externalCephRgwEndpointKey]
-				if err := checkEndpointReachable(rgwEndpoint, 5*time.Second); err != nil {
-					r.Log.Error(err, "RGW endpoint is not reachable.", "RGWEndpoint", rgwEndpoint)
-					return err
-				}
-				extCephObjectStores, err = r.newExternalCephObjectStoreInstances(instance, rgwEndpoint)
-				if err != nil {
-					return err
-				}
+				rgwEndpoint = d.Data[externalCephRgwEndpointKey]
 				// rgw-endpoint is no longer needed in the 'd.Data' dictionary,
 				// and can be deleted
 				// created an issue in rook to add `CephObjectStore` type directly in the JSON output
 				// https://github.com/rook/rook/issues/6165
 				delete(d.Data, externalCephRgwEndpointKey)
```

Comment:
If we delete the rgw endpoint from the d.Data and let's say for some reason the ocs-operator pod got restarted we will loose the end point value.

Reply:
If the ocs-operator is re-started we will re-fecth the data from the secret, using `retrieveExternalSecretData` so all the fields will get back